### ISSUE TITLE
Error: span already finished when running Golden Path test

### ISSUE THREAD

**Issue by oderayi** [mojaloop/project#1087]
**Summary**:
Quoting Service is throwing the following error with some Golden Path tests:

```
[response,api,quotes,sampled] http://0.0.0.0:3002: post /quotes {} 202 (30ms)
(node:31362) UnhandledPromiseRejectionWarning: Error: span already finished
    at Span.<anonymous> (/quoting-service/node_modules/@mojaloop/event-sdk/lib/Span.js:161:39)
    at Generator.next (<anonymous>)
    at /quoting-service/node_modules/@mojaloop/event-sdk/lib/Span.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/quoting-service/node_modules/@mojaloop/event-sdk/lib/Span.js:3:12)
    at Span.finish (/quoting-service/node_modules/@mojaloop/event-sdk/lib/Span.js:159:16)
    at onPreResponse (/quoting-service/node_modules/@mojaloop/central-services-shared/src/util/hapi/plugins/eventPlugin.js:61:12)
    at module.exports.internals.Manager.execute (/quoting-service/node_modules/@hapi/hapi/lib/toolkit.js:41:33)
    at Request._invoke (/quoting-service/node_modules/@hapi/hapi/lib/request.js:339:55)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:31362) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)

```

**Example tests:**
- p2p_money_transfer -> p2p_happy_path SEND_QUOTE -> Send Quote_SEND
- p2p_money_transfer -> p2p_happy_path RECEIVE_QUOTE -> Send Quote_RECEIVE

**Severity**: High
(Low, Medium, High)

**Priority**: Medium
(Critical, Medium, Low)

**Expected Behavior**
No error should be thrown.

**Steps to Reproduce**
  1. Start the quoting service and dependencies
  2. Run the Golden Path test examples
  3. Quoting service console should throw the error above

**Specifications**
  - Component (if known):
  - Version: 8.6.0-snapshot
  - Platform: 
  - Subsystem: Quoting Service
  - Type of testing: Postman
  - Bug found/raised by: @oderayi 

**Notes**:

- Severity when opened: High
- Priority when opened: Medium

**Comment by ggrg** [comment]
@oderayi, please be more specific by including the specific request(s) that is causing the issue.

**Comment by elnyry** [comment]
Just noticed this on quoting-service.. Logs after v8.7.0 release on Dec 12th 4pm GMT, while validating logs after a Golden path test run

```
2019-12-12T16:16:51Z, (1576167411397:d2-870-01-quoting-service-68f7bd959d-4sdrt:25:k42w7x5x:10055) [response,api,quotes,sampled] http://0.0.0.0:3002:  /quotes {}  (34ms) 
(node:25) UnhandledPromiseRejectionWarning: Error: span already finished 
    at Span.<anonymous> (/opt/quoting-service/node_modules/@mojaloop/event-sdk/dist/Span.js:170:39) 
    at Generator.next (<anonymous>) 
    at /opt/quoting-service/node_modules/@mojaloop/event-sdk/dist/Span.js:7:71 
    at new Promise (<anonymous>) 
    at __awaiter (/opt/quoting-service/node_modules/@mojaloop/event-sdk/dist/Span.js:3:12) 
    at Span.finish (/opt/quoting-service/node_modules/@mojaloop/event-sdk/dist/Span.js:168:16) 
    at onPreResponse (/opt/quoting-service/node_modules/@mojaloop/central-services-shared/src/util/hapi/plugins/eventPlugin.js:61:12) 
    at module.exports.internals.Manager.execute (/opt/quoting-service/node_modules/@hapi/hapi/lib/toolkit.js:41:33) 
    at Request._invoke (/opt/quoting-service/node_modules/@hapi/hapi/lib/request.js:339:55) 
    at process._tickCallback (internal/process/next_tick.js:68:7) 
(node:25) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 29) 

```

**Comment by oderayi** [comment]
Bug is now fixed in v9.1.0-snapshot.
